### PR TITLE
Merge stabilization into future-stabilization

### PR DESCRIPTION
**Pull request by vslsnap**
This is an automatically generated pull request from stabilization into future-stabilization.

@dotnet/roslyn-infrastructure:

``` bash
git remote add vslsnap-roslyn "https://github.com/vslsnap/roslyn.git"
git fetch vslsnap-roslyn
git fetch upstream
git checkout merge-stabilization-into-future-stabilization20160615-150226
git reset --hard upstream/future-stabilization
git merge upstream/stabilization
# Fix merge conflicts
git commit
git push vslsnap-roslyn merge-stabilization-into-future-stabilization20160615-150226 --force
```

Once the merge can be made and all the tests pass, you are free to merge the pull request.

### CODE REVIEW

**Comment by vslsnap** [comment]
@dotnet-bot test vsi please

**Comment by jinujoseph** [comment]
@jasonmalinowski @tmat please review

**Comment by cston** [comment]
The changes in https://github.com/dotnet/roslyn/pull/12030/commits/e5a882253616a06307219fdcd5068f63c954f79e should be included if not already.

File: dependencies.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!--
+  Lists dotnet repos containing packages this repo depends on.
```
Comment (jasonmalinowski) [inline dependencies.xml]: Are these correct for what's actually in future-stabilization?

**Comment by jasonmalinowski** [comment]
:+1:

**Pull request merged**